python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/vlan-interfaces.j2

```diff
@@ -315,6 +315,65 @@ interface {{ vlan_interface.name }}
    isis network point-to-point
 {%     endif %}
+{%     if vlan_interface.isis_authentication_mode is arista.avd.defined %}
+{%         set auth_cli = "isis authentication mode" ~ " " ~ vlan_interface.isis_authentication_mode %}
+{%         if vlan_interface.isis_authentication_mode == "sha" and vlan_interface.isis_authentication_mode_sha is arista.avd.defined %}
+{%             if vlan_interface.isis_authentication_mode_sha.both.key_id is arista.avd.defined %}
+{%                 set auth_cli = auth_cli ~ " key-id " ~ vlan_interface.isis_authentication_mode_sha.both.key_id %}
+{%                 if vlan_interface.isis_authentication_mode_sha.both.rx_disabled is arista.avd.defined %}
+{%                     set auth_cli = auth_cli ~ " rx-disabled" %}
+{%                 endif %}
+   {{ auth_cli }}
+{%             else %}
+{%                 if vlan_interface.isis_authentication_mode_sha.level_1.key_id is arista.avd.defined %}
+{%                     if vlan_interface.isis_authentication_mode_sha.level_1.rx_disabled is arista.avd.defined %}
+{%                         set auth_cli = auth_cli ~ " rx-disabled" %}
+{%                     endif %}
+{%                     set auth_cli = auth_cli ~ " key-id " ~ vlan_interface.isis_authentication_mode_sha.level_1.key_id ~ " level-1" %}
+   {{ auth_cli }}
+{%                 endif %}
+{%                 if vlan_interface.isis_authentication_mode_sha.level_2.key_id is arista.avd.defined %}
+{%                     if vlan_interface.isis_authentication_mode_sha.level_2.rx_disabled is arista.avd.defined %}
+{%                         set auth_cli = auth_cli ~ " rx-disabled" %}
+{%                     endif %}
+{%                     set auth_cli = auth_cli ~ " key-id " ~ vlan_interface.isis_authentication_mode_sha.level_2.key_id ~ " level-2" %}
+   {{ auth_cli }}
+{%                 endif %}
+{%             endif %}
+{%         endif %}
+{%         if vlan_interface.isis_authentication_mode == "shared-secret" and vlan_interface.isis_authentication_mode_shared_secret is arista.avd.defined %}
+{%             if vlan_interface.isis_authentication_mode_shared_secret.both.profile is arista.avd.defined and vlan_interface.isis_authentication_mode_shared_secret.both.algorithm is arista.avd.defined %}
+{%                 set auth_cli = auth_cli ~ " profile " ~  vlan_interface.isis_authentication_mode_shared_secret.both.profile ~ " algorithm " ~ vlan_interface.isis_authentication_mode_shared_secret.both.algorithm %}
+{%                 if vlan_interface.isis_authentication_mode_shared_secret.both.rx_disabled is arista.avd.defined %}
+{%                     set auth_cli = auth_cli ~ " rx-disabled " %}
+{%                 endif %}
+   {{ auth_cli }}
+{%             else %}
+{%                 if vlan_interface.isis_authentication_mode_shared_secret.level_1.profile is arista.avd.defined and vlan_interface.isis_authentication_mode_shared_secret.level_1.algorithm is arista.avd.defined %}
```

Comment:
```suggestion
{%             elif vlan_interface.isis_authentication_mode_shared_secret.level_1.profile is arista.avd.defined and vlan_interface.isis_authentication_mode_shared_secret.level_1.algorithm is arista.avd.defined %}
```

correct the indentation after applying this

Reply:
If `both` is not set then only level1 and level2 are configured. So added else condition after both.

Reply:
Okay, it will get converted in CLI (as both)